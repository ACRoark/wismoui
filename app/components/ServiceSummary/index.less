@import (reference) '../../styles/themes/index.less';

@service-summary-cls-prefix: ~'@{dte-prefix}-service-summary';

.@{service-summary-cls-prefix} {
  border: 1px solid @dte-gray-400;
  display: flex;
  margin: 5px 0;
  padding: 30px;

  &-field-container {
    margin-right: 12px;
    width: 50%;
  }

  @media screen and (max-width: 420px) {
    flex-direction: column;

    &-field-container {
      width: 100%;
    }

    &-field-container:nth-child(1) {
      padding-bottom: 20px;
    }
  }
}
