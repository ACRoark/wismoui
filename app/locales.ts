export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
