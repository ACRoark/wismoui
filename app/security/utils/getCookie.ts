const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
