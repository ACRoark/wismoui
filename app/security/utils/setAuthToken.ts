import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
