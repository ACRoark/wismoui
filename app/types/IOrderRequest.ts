import IOrderRequestBase from './IOrderRequestBase';
import IOrderRequestStatusUpdate from './IOrderRequestStatusUpdate';
import IServiceOrderEvent from './IServiceOrderEvent';
import IVerificationEvent from './IVerificationEvent';

interface IOrderRequest extends IOrderRequestBase {
  contactPhoneNumber: string;
  orderRequestStatusUpdates: IOrderRequestStatusUpdate[];
  serviceOrderEvents: IServiceOrderEvent[];
  verificationEvents: IVerificationEvent[];
}

export default IOrderRequest;
