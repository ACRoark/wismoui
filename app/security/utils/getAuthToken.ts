import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
