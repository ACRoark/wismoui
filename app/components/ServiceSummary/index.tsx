import * as React from 'react';

import AddressFormField from 'components/AddressField';
import ProductsFormField from 'components/ProductsListField';

import IOrderRequest from 'types/IOrderRequest';

import './index.less';

interface IServiceSummaryProps {
  serviceRequest: IOrderRequest;
}

const ServiceSummary: React.FC<IServiceSummaryProps> = (props: IServiceSummaryProps): React.ReactElement => {
  const { address, products } = props.serviceRequest;

  return (
    <div className="dte-wismo-service-summary">
      <div className="dte-wismo-service-summary-field-container">
        <AddressFormField address={address} />
      </div>
      <div className="dte-wismo-service-summary-field-container">
        <ProductsFormField products={products} />
      </div>
    </div>
  );
};

export default ServiceSummary;
