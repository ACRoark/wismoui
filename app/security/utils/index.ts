import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
