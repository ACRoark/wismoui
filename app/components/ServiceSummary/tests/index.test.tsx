import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import ServiceSummary from 'components/ServiceSummary';
import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import IOrderRequest from 'types/IOrderRequest';

import fakeServiceRequest from './fakeServiceRequest';

const serviceRequestNoProducts = {
  ...fakeServiceRequest,
  products: [],
};

const store = createMockStore();

const expectNoConsoleErrors = (language: string, order: IOrderRequest): void => {
  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(<Provider store={store}><ServiceSummary serviceRequest={order} /></Provider>, language);

  expect(spy).not.toHaveBeenCalled();
};

const testServiceSummarySnapshot = (language: string, order: IOrderRequest): void => {
  const tree = createSnapshotWithIntl(<Provider store={store}><ServiceSummary
    serviceRequest={order} /></Provider>, language);

  expect(tree).toMatchSnapshot();
};

describe('<ServiceSummary />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered with a valid order in ${language}`, (): void =>
      expectNoConsoleErrors(language, fakeServiceRequest));

    it(`should not log errors in console when rendered with an order that has no products in ${language}`, (): void =>
      expectNoConsoleErrors(language, fakeServiceRequest));

    it(`should render correctly when given a valid order, and current language is ${language}`, (): void =>
      testServiceSummarySnapshot(language, fakeServiceRequest));

    it(`should render correctly when given an order with no products, and current language is ${language}`, (): void =>
      testServiceSummarySnapshot(language, serviceRequestNoProducts));
  });
});
