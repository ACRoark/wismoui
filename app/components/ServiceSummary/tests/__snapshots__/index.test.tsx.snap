// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceSummary /> should render correctly when given a valid order, and current language is en 1`] = `
<div
  className="dte-wismo-service-summary"
>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        Address
      </label>
      <span
        className="dte-wismo-base-field-children"
      >
        <address
          className="dte-wismo-address"
        >
          123 Main
Bloomfield, MI 48301
        </address>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        List of Service(s)
      </label>
      <span
        className="dte-wismo-base-field-children"
      >
        Residential Electric Interruptible Air Conditioning
        , 
        Residential Electric Service
        , 
        Natural Gas Residential Heating
      </span>
    </div>
  </div>
</div>
`;

exports[`<ServiceSummary /> should render correctly when given a valid order, and current language is es 1`] = `
<div
  className="dte-wismo-service-summary"
>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        Dirección
      </label>
      <span
        className="dte-wismo-base-field-children"
      >
        <address
          className="dte-wismo-address"
        >
          123 Main
Bloomfield, MI 48301
        </address>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        Lista de servicio(s)
      </label>
      <span
        className="dte-wismo-base-field-children"
      >
        Aire acondicionado eléctrico interrumpible residencial
        , 
        Servicio Eléctrico Residencial
        , 
        Calefacción Residencial de Gas Natural
      </span>
    </div>
  </div>
</div>
`;

exports[`<ServiceSummary /> should render correctly when given an order with no products, and current language is en 1`] = `
<div
  className="dte-wismo-service-summary"
>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        Address
      </label>
      <span
        className="dte-wismo-base-field-children"
      >
        <address
          className="dte-wismo-address"
        >
          123 Main
Bloomfield, MI 48301
        </address>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        List of Service(s)
      </label>
      <span
        className="dte-wismo-base-field-children"
      />
    </div>
  </div>
</div>
`;

exports[`<ServiceSummary /> should render correctly when given an order with no products, and current language is es 1`] = `
<div
  className="dte-wismo-service-summary"
>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        Dirección
      </label>
      <span
        className="dte-wismo-base-field-children"
      >
        <address
          className="dte-wismo-address"
        >
          123 Main
Bloomfield, MI 48301
        </address>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-service-summary-field-container"
  >
    <div
      className="dte-wismo-base-field-section"
    >
      <label
        className="dte-wismo-base-field-label"
      >
        Lista de servicio(s)
      </label>
      <span
        className="dte-wismo-base-field-children"
      />
    </div>
  </div>
</div>
`;
